ts/utils/dates.ts

```diff
@@ -161,12 +161,13 @@ export const isExpired = (
 ): Either<Error, boolean> => {
   const maybeMonth = decodeCreditCardMonth(expireMonth);
   const maybeYear = decodeCreditCardYear(expireYear);
-
   return maybeYear
     .chain(year =>
       maybeMonth.map(month => {
         const now = new Date();
-        return year < now.getFullYear() || month < now.getMonth() + 1;
+        const nowYearMonth = new Date(now.getFullYear(), now.getMonth());
+        const cardExpirationDate = new Date(year, month - 1);
+        return nowYearMonth > cardExpirationDate;
```

Comment:
```suggestion
        const nowYearMonth = new Date(now.getFullYear(), now.getMonth()+1);
        const cardExpirationDate = new Date(year, month);
        return nowYearMonth > cardExpirationDate;
```
**🎯 suggestion:** ‏In order to compare the effective dates (the result is the same because the previous implementation shifted both dates back one month).

Reply:
Sounds good!
refactored here https://github.com/pagopa/io-app/pull/3714/commits/37fb5d6b4a088ab33754191efb4eca1c83f038ec